io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/reader/internal/ShapeInstanceReader.java

```diff
@@ -382,6 +399,143 @@ private static ChildDefinition<?> getChild(TypeDefinition schemaType, QName prop
 		return result;
 	}
 
+	/**
+	 * Download Shapefiles from the given URL if a .fix file exists at the URL.
+	 * Otherwise, IOException related to `Read Channel is not a File Channel` is
+	 * thrown when reading indexes.
+	 *
+	 * @param loc the URI location of the shapefile
+	 * @param reporter the reporter to report errors
+	 * @return the temporary directory where the shapefiles are downloaded, or
+	 *         <code>null</code> if no files were downloaded
+	 */
+	private File downloadShpFilesIfNeeded(URI loc, IOReporter reporter) {
+
+		try {
+			URL locUrl = loc.toURL();
+			// extract baseName of the file so that we can download all the files if they
+			// exist.
+			String baseName = getBaseName(locUrl);
+
+			// create URL for .fix file from whatever URI is passed
+			URL fixFileUrl = URI.create(baseName.concat(ShpFileType.FIX.extensionWithPeriod))
+					.toURL();
+			boolean fixFileExists = checkFileExistence(fixFileUrl, reporter);
+			if (fixFileExists) {
+				reporter.info(
+						"Valid URL {0} is provided and also provides FIX file. Therefore, downloading all the available files to proceed with Shapefile processing ",
+						loc.toString());
+				// proceed to download all the files from the URL
+				File tempDir = Files.createTempDirectory("shp_from_url_").toFile();
+				shutdownHookToDeleteTempDir(tempDir, reporter);
```

Comment:
I think it would be nice to add a comment why the shutdown hook is used and we don't use any mechanism that can do cleanup earlier. For the next developer that looks at the code and wonders.